plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -623,4 +640,99 @@ private static void validateOrcBloomFilterColumns(ConnectorTableMetadata tableMe
             throw new TrinoException(INVALID_TABLE_PROPERTY, format("Orc bloom filter columns %s not present in schema", Sets.difference(ImmutableSet.copyOf(orcBloomFilterColumns), allColumns)));
         }
     }
+
+    public static Optional<String> getLatestMetadataLocation(TrinoFileSystem fileSystem, String location)
+    {
+        Set<String> latestMetadataLocations = new HashSet<>();
+        String metadataDirectoryLocation = format("%s/%s", stripTrailingSlash(location), METADATA_FOLDER_NAME);
+        try {
+            int latestMetadataVersion = -1;
+            FileIterator fileIterator = fileSystem.listFiles(metadataDirectoryLocation);
+            while (fileIterator.hasNext()) {
+                FileEntry fileEntry = fileIterator.next();
+                if (fileEntry.path().endsWith(METADATA_FILE_EXTENSION)) {
+                    OptionalInt version = parseVersion(fileEntry.path());
+                    if (version.isPresent()) {
+                        if (version.getAsInt() > latestMetadataVersion) {
+                            latestMetadataVersion = version.getAsInt();
+                            latestMetadataLocations.clear();
+                            latestMetadataLocations.add(fileEntry.path());
+                        }
+                        else if (version.getAsInt() == latestMetadataVersion) {
+                            latestMetadataLocations.add(fileEntry.path());
+                        }
+                    }
+                }
+            }
+            if (latestMetadataLocations.isEmpty()) {
+                throw new TrinoException(ICEBERG_INVALID_METADATA, "No metadata file exists at location: " + metadataDirectoryLocation);
+            }
+            if (latestMetadataLocations.size() > 1) {
+                throw new TrinoException(ICEBERG_INVALID_METADATA, format("More than one latest metadata file found at location: %s, latest metadata files are %s",
+                        metadataDirectoryLocation, latestMetadataLocations));
+            }
+        }
+        catch (IOException e) {
+            throw new TrinoException(ICEBERG_FILESYSTEM_ERROR, "Failed checking table's location: " + location, e);
+        }
+        return Optional.of(getOnlyElement(latestMetadataLocations.iterator()));
```

Comment:
.iterator is redundant

```suggestion
        return Optional.of(getOnlyElement(latestMetadataLocations));
```

Reply:
Removing `.iterator` gives below compile time error.
 
Required type:
`Iterator <T>`
Provided:
`List  <String>`
reason: no instance(s) of type variable(s) T exist so that `List<String>` conforms to `Iterator<T>`

Reply:
you can replace `Iterators.getOnlyElement` with `Iterables.getOnlyElement`

Reply:
Addressed